<!-- Demonstrating PID controller with the help of turtlesim -->
<!-- with the fine tuned PID gains, the robot should move to the x0 location in the optimum time without oscillation -->

<launch>
    <!-- Arguments -->
    <arg name="x0"  default="7.0"/>
    <arg name="m"   default="1.0"/>
    <arg name="kpd" default="2"/>
    <arg name="kdd" default="0.04"/>
    <arg name="kid" default="0.000005"/>

    <node name="turtlesim_node"   type="turtlesim_node"   pkg="turtlesim"/>
    <node name="pidLinController" type="pidLinController" pkg="roslearning" output="screen">
        <param name="x0"  value="$(arg x0)"  type="double"/>
        <param name="m"   value="$(arg m)"   type="double"/>
        <param name="kpd" value="$(arg kpd)" type="double"/>
        <param name="kid" value="$(arg kid)" type="double"/>
        <param name="kdd" value="$(arg kdd)" type="double"/>
    </node>
</launch>